### DISCUSSION TITLE
Micro apps

### DISCUSSION THREAD

**Discussion by sp8cefox** [mikedilger/gossip#904] [Ideas]
Have you though about making gossip into more of a platform where micro apps can be turned on or off?
The rationale: Many micro apps could be build using the gossip-lib on it own, as stand alone applications.
However in many cases it might be handy to incubate into gossip-bin and thereby let the micro app be part of the users' everyday use of Gossip. Yet, some users may not want certain features so it would be fair to let them go lightweight.

**Comment by mikedilger** [comment]
I separated gossip-lib from gossip-bin because numerous people were suggesting UI/UX ideas for gossip and those ideas were going in multiple different directions.  I basically said "Here:  the library is now separate.  If you want a different kind of UI, go build it yourself".

So you could build a very different kind of gossip program that still used it's internal library.

As for "micro apps", gossip is not a platform nor do I want it to be a platform.  The whole point of gossip is to provide a secure nostr client that isn't vulnerable to web-stack vulnerabilities.  Adding any kind of plug-in system would seriously weaken that selling point.
